database/migrations/default/1750435376415_ems_person_match_auto/up.sql

```diff
@@ -0,0 +1,131 @@
+--
+-- add person_match_status column
+--
+alter table ems__incidents
+    add column person_match_status text default 'unmatched'
+    check (
+        person_match_status in (
+            'unmatched',
+            'matched_by_automation',
+            'matched_by_manual_qa',
+            'multiple_matches_by_automation'
+        )
+    ),
+    add column matched_person_ids integer [];
+
+comment on column ems__incidents.person_match_status is 'The status of the CR3 person record match';
+comment on column ems__incidents.matched_person_ids is 'The IDs of the CR3 person records that were found to match this record';
+
+--
+-- Add person_match_status to this trigger
+--
+CREATE OR REPLACE FUNCTION public.ems_update_person_crash_id()
+ RETURNS trigger
+ LANGUAGE plpgsql
+AS $function$
+DECLARE
+    matching_person_record RECORD;
+BEGIN
+    IF NEW.person_id is null THEN
+        NEW.person_match_status = 'unmatched';  -- <- new line
+        return NEW;
+    END IF;
+    -- Find matching person record
+    SELECT id, crash_pk INTO matching_person_record 
+    FROM people_list_view 
+    WHERE id = NEW.person_id;
+
+    NEW.crash_pk = matching_person_record.crash_pk;
+    NEW.crash_match_status = 'matched_by_manual_qa';
+    NEW.person_match_status = 'matched_by_manual_qa';  -- <- new line
+    return NEW;
+END;
+$function$;
+
+--
+-- Create trigger function to match person record 
+--
+CREATE OR REPLACE FUNCTION public.ems_person_ems_match()
+ RETURNS trigger
+ LANGUAGE plpgsql
+AS $function$
+DECLARE
+    other_matching_ems_id INTEGER;
+    matching_person_ids INTEGER[];
+BEGIN
+    IF NEW.person_match_status = 'matched_by_manual_qa' 
+    or NEW.pcr_patient_age is NULL
+    or NEW.pcr_patient_gender is NULL
+    or NEW.pcr_patient_race is NULL
+    THEN
+        -- do nothing
+        return NEW;
+    END IF;
+
+    IF NEW.crash_pk is NULL THEN
+        new.person_match_status = 'unmatched';
+        new.matched_person_ids = NULL;
+        return NEW;
+    END IF;
+
+    --
+    -- Check to see if there are other EMS patients with the same demographics as NEW
+    --
+    select id from ems__incidents e where
+    e.incident_number = NEW.incident_number
+    and e.id !=  NEW.id
+    and e.pcr_patient_age = NEW.pcr_patient_age
+    and e.pcr_patient_race = NEW.pcr_patient_race
+    and e.pcr_patient_gender = NEW.pcr_patient_gender
+    limit 1
+    into other_matching_ems_id;
+
+    --
+    -- Stop if multiple identical PCRs 
+    -- todo: we could potentially try to match multiple records to multiple people?
```

Comment:
we could also try to match on person occupant position. i'm inclined to leave this out of scope.

Reply:
i shared [this article](https://sophiebits.com/2025/07/21/todos-arent-for-doing) in dev sync recently, and it made me think of your style of todos. good stuff!